bin/s2nc.c

```diff
@@ -605,7 +631,25 @@ int main(int argc, char *const *argv)
 
         GUARD_EXIT(s2n_connection_free_handshake(conn), "Error freeing handshake memory after negotiation");
 
-        if (echo_input == 1) {
+        bool run_perf = false;
+        bool run_echo = false;
```

Comment:
I mostly did it like this so it would be easier to add new functionality in the future. I also like that each branch is explicitly handled in the `if` rather than setting it above. The more correct thing to do if we get an ALPN that we don't support is reject the connection.